File: 7_initialization/javatests/src/test/java/javatests/OtherConstructors.java
package javatests;

public class OtherConstructors {

    public OtherConstructors() {
        this("default value");
    }

    public OtherConstructors(String someString) {
        this.someString = someString;
    }

    private String someString;

    public String getSomeString() {
        return someString;
    }
}

File: 7_initialization/dotnet/tests/OtherConstructors.cs
namespace tests
{
    public class OtherConstructors
    {
        public string SomeString { get; }

        public OtherConstructors() : this("default value")
        {}

        public OtherConstructors(string _someString)
        {
            SomeString = _someString;
        }
    }
}